### ISSUE TITLE
[Console] Enhanced stack trace memory problem

### ISSUE THREAD

**Issue by mvrhov** [symfony/symfony#21943]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Symfony version  | master

This #21705 seems to introduce a memory hog. 
In most occasions you are greeted with php terminating script due to memory exhaustion (128M) instead of getting the stack dump. 
I have run the command with time to see the top memory consumption and it was around 150M-160M.
Running same command with 3.2, the memory used is around 50M

ping @lyrixx

**Comment by lyrixx** [comment]
hello @mvrhov

Do you have a simple reproducer for this bug?

**Comment by mvrhov** [comment]
Just make a command that throws an exception? The more data in stacktrace the easier this can be spotted.

**Comment by lyrixx** [comment]
I'm not sure to fully understand your issue.

Your issue is: When a command throw an exception, Symfony use more memory than before?

**Comment by mvrhov** [comment]
I had a command where I didn't catch the exception. Instead of getting the stacktrace or just the message that the exception happened, the PHP was killed because of OOM.

**Comment by lyrixx** [comment]
OK, Thanks, I understand now.

It will not be easy to reproduce. But indeed the VarCloners take more resources than before.

maybe @nicolas-grekas has an idea for that.

**Comment by lyrixx** [comment]
I just tried to reproduce your issue:

1. I used symfony-standard.
1. I created a command, and in this command I just throw an exception.
1. in `bin/console` I added: `register_shutdown_function(function() { echo memory_get_peak_usage()."\n"; });`

I ran this command on master and on 3.2. Symfony uses `2847624` on master and `2746480` on 3.2. So the delta is `101144`...

So for me there are no issue. 

Could you fork the SE and put a reproducer in it? If you can not provide a reproducer, I will have to close this issue.

Thanks.

**Comment by nicolas-grekas** [comment]
@mvrhov can you maybe provide a blackfire memory profile?

**Comment by mvrhov** [comment]
Here you go: https://blackfire.io/profiles/compare/105e3b15-c8bf-4b86-afdc-4c80ce1907d2...44b979a8-4abf-4897-9902-f9100d5e3228/graph?settings%5Bdimension%5D=wt&settings%5Bdisplay%5D=landscape&settings%5BtabPane%5D=nodes&selected=&callname=main()

**Comment by lyrixx** [comment]
@mvrhov Don't forget the reproducer please.

**Comment by nicolas-grekas** [comment]
The profile tells that AbstractCloner calls ErrorHandler 2501 times. Can you give us some messages that get there (the most frequent?) They should be in the logs, or maybe you can just "echo $message"  in AbstractCloner just before the `call_user_func($this->prevErrorHandler,...` call.

**Comment by lyrixx** [comment]
Ok, I found a way to reproduce it.

SE on master + 
```diff
diff --git a/app/config/config_dev.yml b/app/config/config_dev.yml
index 3c86425..eb0e83a 100644
--- a/app/config/config_dev.yml
+++ b/app/config/config_dev.yml
@@ -20,7 +20,7 @@ monolog:
             channels: ['!event']
         console:
             type:   console
-            channels: ['!event', '!doctrine', '!console']
+            channels: ['!event', '!doctrine']
         # uncomment to get logging in your browser
         # you may have to allow bigger header sizes in your Web server configuration
         #firephp:
diff --git a/bin/console b/bin/console
index a908aaa..8d91e56 100755
--- a/bin/console
+++ b/bin/console
@@ -23,6 +23,8 @@ if ($debug) {
     Debug::enable();
 }
 
+register_shutdown_function(function() { echo "\n".memory_get_peak_usage()."\n"; });
+
 $kernel = new AppKernel($env, $debug);
 $application = new Application($kernel);
 $application->run($input);
diff --git a/src/AppBundle/Command/LongRunningCommand.php b/src/AppBundle/Command/LongRunningCommand.php
new file mode 100644
index 0000000..2885116
--- /dev/null
+++ b/src/AppBundle/Command/LongRunningCommand.php
@@ -0,0 +1,25 @@
+<?php
+
+namespace AppBundle\Command;
+
+
+use Psr\Log\LogLevel;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class LongRunningCommand extends ContainerAwareCommand
+{
+    protected function configure()
+    {
+        $this
+            ->setName('app:exception')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        throw new \Exception("Error Processing Request", 1);
+    }
+}
```

**Comment by nicolas-grekas** [comment]
Could be fixed by #21973, please confirm.

**Comment by mvrhov** [comment]
Yep, down to 21.3M according to blackfire
